omod/src/main/java/org/openmrs/module/idgen/rest/resource/AutoGenerationOptionResource.java

```diff
@@ -0,0 +1,245 @@
+		if (errors.size() > 0) {
+			throw new ValidationException("The values of the following inputs are missing: " + errors.toString());
+		}
+		IdentifierSource identifierSource = identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString());
+		if (identifierSource == null) {
+			errors.add("source");
+		}
+		PatientIdentifierType patientIdentifierType = Context.getPatientService()
+		        .getPatientIdentifierTypeByUuid(identifierTypeUuid.toString());
+		if (patientIdentifierType == null) {
+			errors.add("IdentifierType");
+		}
+		if (errors.size() > 0) {
+			throw new ValidationException("The values of the following inputs are invalid: " + errors.toString());
+		}
+		Location location = null;
+		if (locationUuid != null && StringUtils.isNotBlank(identifierTypeUuid.toString())) {
+			location = Context.getLocationService().getLocationByUuid(locationUuid.toString());
+		}
+		if (location == null) {
+			location = Context.getLocationService()
+			        .getLocationByUuid(Context.getLocationService().getDefaultLocation().getUuid());
+		}
+		AutoGenerationOption autoGenerationOption = new AutoGenerationOption();
+		autoGenerationOption.setSource(identifierSource);
+		autoGenerationOption.setIdentifierType(patientIdentifierType);
+		autoGenerationOption.setManualEntryEnabled(this.parseBoolean(manualEntryEnabled.toString()));
+		autoGenerationOption.setAutomaticGenerationEnabled(this.parseBoolean(automaticallyEnabled.toString()));
+		autoGenerationOption.setLocation(location);
+		return (SimpleObject) ConversionUtil.convertToRepresentation(save(autoGenerationOption), Representation.FULL);
+		
+	}
+	
+	@Override
+	public SimpleObject update(String uuid, SimpleObject postBody, RequestContext context) throws ResponseException {
+		int id = 0;
+		try {
+			id = Integer.parseInt(uuid);
+		}
+		catch (NumberFormatException e) {
+			throw new ValidationException("The uuid enterd is invalid: " + e.getMessage());
+		}
+		AutoGenerationOption autoGenerationOption = identifierSourceService.getAutoGenerationOption(id);
+		if (autoGenerationOption == null) {
+			throw new ObjectNotFoundException();
+		}
+		Object manualEntryEnabled = (postBody.get("manualEntryEnabled"));
+		Object automaticGenerationEnabled = postBody.get("automaticGenerationEnabled");
+		Object sourceUuid = postBody.get("source");
+		Object locationUuid = postBody.get("location");
+		IdentifierSource identifierSource = (sourceUuid != null && StringUtils.isNotBlank(sourceUuid.toString())
+		        && identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString()) != null)
+		                ? identifierSourceService.getIdentifierSourceByUuid(sourceUuid.toString())
+		                : null;
+		Location location = (locationUuid != null && StringUtils.isNotBlank(locationUuid.toString())
+		        && Context.getLocationService().getLocationByUuid(locationUuid.toString()) != null)
+		                ? Context.getLocationService().getLocationByUuid(locationUuid.toString())
```

Comment:
Here the same comments apply as in the create method above.

Reply:
Just take a look at how you did your create method and you will notice that you have done this differently.